.bazelrc

```diff
@@ -31,11 +31,38 @@ build:wasm32 --symlink_prefix=bazel-wasm-
 build:clang --crosstool_top=//toolchain:clang
 build:clang --host_crosstool_top=@bazel_tools//tools/cpp:toolchain
 build:clang --cpu=k8
-build:clang --copt=-Wno-error=sign-compare
-
+# Force the usage of stack protectors for all functions.
+build:clang --cxxopt="-fstack-protector-all"
+# Pretty colours.
+build:clang --cxxopt="-fcolor-diagnostics"
+# Don't omit the frame pointer, useful for debug.
+build:clang --cxxopt="-fno-omit-frame-pointer"
+# Enable all warnings.
+build:clang --cxxopt="-Wall"
+# Enable extra warnings.
+build:clang --cxxopt="-Wextra"
+# Simple but useful static checking to detect potential race conditions.
+build:clang --cxxopt="-Wthread-safety"
+# Treat warnings as errors.
```

Comment:
How [fragile](https://embeddedartistry.com/blog/2017/05/22/werror-is-not-your-friend/) is this?  If we happen to update a dependency that has a new warning, or if we upgrade Clang version to one with more warnings on existing code, isn't this going to break the build?

Reply:
Talked about this. Let's keep it like this for a bit and if it causes issues we drop it.